src/prefect_server/graphql/schema/agents.graphql

```diff
@@ -0,0 +1,76 @@
+extend type Mutation {
+  # ------------------------------------------------------
+  # agents
+  # ------------------------------------------------------
+
+  "Register an agent instance"
+  register_agent_instance(input: register_agent_instance_input!): agent_instance_id_payload
+
+  "Delete an agent instance"
+  delete_agent_instance(input: delete_agent_instance_input!): success_payload
+
+  "Create a new agent"
+  create_agent(input: create_agent_input!): agent_id_payload
+
+  "Delete an agent"
+  delete_agent(input: delete_agent_input!): success_payload
+
+  "Update an agent's config"
+  set_agent_config(input: set_agent_config_input!): success_payload
+
+}
+
+# ------------------------------------------------------
+# agent inputs
+# ------------------------------------------------------
+
+input register_agent_instance_input {
+  "the tenant"
+  tenant_id: UUID!
+  "instance labels"
+  labels: [String]!
+  "agent id"
+  agent_id: UUID
+  "instance type"
+  type: String
+  "instance name"
+  name: String
+}
+
+input delete_agent_instance_input {
+  "The ID of the agent instance to delete"
+  agent_instance_id: UUID!
+}
+
+input create_agent_input {
+  "The tenant"
+  tenant_id: UUID!
+  "Agent name"
+  name: String
+  "Optional agent configuration"
+  config: JSON
+}
+
+input delete_agent_input {
+  "The ID of the agent to delete"
+  agent_id: UUID!
+}
+
+input set_agent_config_input {
+  "The ID of the agent to update"
+  agent_id: UUID!
+  "A config to set for this agent"
+  config: JSON
```

Comment:
This needs to be required, because 1) it is accessed as `input['config']` in the resolver and 2) null configs are not allowed in the db

```suggestion
  config: JSON!
```